static/js/invite.js

```diff
@@ -191,12 +193,31 @@ export function launch() {
     });
 }
 
+function valid_to() {
+    var time_valid = Number.parseFloat($("#expires_in").val());
+    if (!time_valid) {
+        return 'Never expires';
+    }
+    Date.prototype.addDays = function(days) {
+        var date = new Date(this.valueOf());
+        date.setDate(date.getDate() + days);
+        return date;
+    }
+    var date = new Date();
+    var valid_to = date.addDays(time_valid);
+    var dd = String(valid_to.getDate()).padStart(2, '0');
+    var mm = String(valid_to.getMonth() + 1).padStart(2, '0');
+    var yyyy = valid_to.getFullYear();
+    return "Expires on " + dd + '/' + mm + '/' + yyyy;
+}
```

Comment:
This logic doesn't support internationalization.  I think it's likely that you can make this work by calling a function from `timerender.js` or `date-fns`, and then just have a string "Expires on {date}" | "Never expires" as the string that gets internationalized.

https://zulip.readthedocs.io/en/latest/translating/internationalization.html is helpful background reading.

Reply:
Yes, I did not know how to solve it at first, but after reviewing your suggestion I think that date-fns is the best alternative for this. I followed the internationalization guide and think this should now work.

Reply:
Great!